addon/components/vertical-collection/component.js

```diff
@@ -181,6 +181,7 @@ const VerticalCollection = Component.extend({
         run(() => {
           const items = this.get('items');
           const keyPath = this.get('key');
+          if (!items || items.length === 0) { return; }
```

Comment:
we should also not fire below if index exceeds the bounds of length. Otherwise this LGTM